### ISSUE TITLE
Add nullable annotations

### ISSUE THREAD

**Issue by TravisNorwood** [FirelyTeam/fhir-net-api#1417]
Add C# 8.0 nullable annotations to the API. Now that the project uses C# 8.0, the classes can be marked with nullable annotations (adding a ? after a parameter or return type to indicate that it could be null). This would be a tremendous help to users of the API to help increase quality by reducing bugs like null reference exceptions.

**Comment by ewoutkramer** [comment]
I fully agree, and I have started doing it.  However, with 1.700 classes and 300.000 lines of code, this is going to take a while....

**Comment by ewoutkramer** [comment]
Basically, when I am adding something new to the API or doing a major refactor, I'll be adding annotations....

**Comment by ewoutkramer** [comment]
@marcovisserFurore @mmsmits @brianpos - maybe -as major committers - we could agree on using such a step-by-step approach? What do you think?

**Comment by marcovisserFurore** [comment]
A step-by-step approach will be fine.  I am not looking forwards to scan 300.000 lines of code in one go.

**Comment by ewoutkramer** [comment]
Ok, let's do that then!  You'll start seeing them from version 2.0 onwards!
